crates/fluvio/src/producer/memory_batch.rs

```diff
@@ -10,14 +10,14 @@ use super::*;
 
 pub struct MemoryBatch {
     compression: Compression,
-    write_limit: usize,
+    write_limit: Option<usize>,
```

Comment:
why this is optional?

Reply:
For cases when we don't want a limit, for example when a single record is larger than `batch_size`. 

I can also back to the strategy of returning the record_size and try again with the final record_size as batch_size